### PR TITLE
Add TestingClock

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

File: testing/src/main/java/io/airlift/testing/TestingClock.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public final class TestingClock
```
Comment (kokosing) [inline testing/src/main/java/io/airlift/testing/TestingClock.java]: Can you please add javadoc?

File: testing/src/main/java/io/airlift/testing/TestingClock.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        extends Clock
+{
+    private final ZoneId zone;
+    private Instant instant;
+
+    public TestingClock()
+    {
+        this(ZoneOffset.UTC);
```
Comment (kokosing) [inline testing/src/main/java/io/airlift/testing/TestingClock.java]: Why do you care about time zone? Is it going to be stored or compared between different time zones?

File: testing/src/main/java/io/airlift/testing/TestingClock.java (hunk 1 above)
Comment (findepi) [reply testing/src/main/java/io/airlift/testing/TestingClock.java]: I don't think it's necessary. What would i write beyond "Testing Clock"?

File: testing/src/main/java/io/airlift/testing/TestingClock.java (hunk 2 above)
Comment (findepi) [reply testing/src/main/java/io/airlift/testing/TestingClock.java]: See `Clock` "interface" (ie what methods are abstract)

**Pull request merged**